pkgs/tools/graphics/agi/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, fetchzip
+, autoPatchelfHook
+, makeWrapper
+, makeDesktopItem
+, wrapGAppsHook
+, gobject-introspection
+, jre
+, androidenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "agi";
+  version = "1.1.0-dev-20210409";
+
+  src = fetchzip {
+    url = "https://github.com/google/agi-dev-releases/releases/download/v${version}/agi-${version}-linux.zip";
+    sha256 = "1862qp03qrbv1xxjp0c0v4ngd3lk8ak91m8cch6gcbjylrga2ma0";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    wrapGAppsHook
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib}
+    cp ./{agi,gapis,gapir,gapit,device-info} $out/bin
+    cp lib/gapic.jar $out/lib
+
+    wrapProgram $out/bin/agi \
+      --add-flags "--vm ${jre}/bin/java" \
+      --add-flags "--jar $out/lib/gapic.jar" \
+      --add-flags "--adb ${androidenv.androidPkgs_9_0.platform-tools}/bin/adb"
+
+    for i in 16 32 48 64 96 128 256 512 1024; do
+      install -D ${src}/icon.png $out/share/icons/hicolor/''${i}x$i/apps/agi.png
+    done
+    cp -r ${makeDesktopItem {
+      name = pname;
+      desktopName = "Android GPU Inspector";
+      exec = "agi";
+      icon = "agi";
+      type = "Application";
+      categories = "Development;Debugger;Graphics;3DGraphics";
+      terminal = "false";
+    }}/share/applications $out/share
```

Comment:
Please use copyDesktopItems.

Reply:
I'm not very familiar with this function, but from grepping through nixpkgs i gather you should have an `let` block like this:
```nix
let
  desktopItem = makeDesktopItem {
  };
in stdenv.mkDerivation rec {
```

Which IMO looks way worse, is there any guidelines on this? If no, I'd prefer to leave it as-is.

Reply:
You can also inline it like https://github.com/NixOS/nixpkgs/blob/7248f89da2dab8bca3a9f4f11d0407d0c8e25e89/pkgs/applications/blockchains/mycrypto/default.nix#L20.

Reply:
Sure, that makes sense.

I have not been able to get the function working, as with everything I've tried it simply ends up not copying over the desktop item.

My WIP derivation using `copyDesktopItems`: https://gist.github.com/IvarWithoutBones/7671f660157b6ba8206a5f787122594f
```
~/nix/nixpkgs/pkgs/tools/graphics/agi > ls ./result/share/
icons
```

Reply:
You are not inheriting desktopItem. https://github.com/NixOS/nixpkgs/blob/ffd29c76f682ffdec5e9a2b7fb81c3f07f679361/pkgs/applications/misc/dbeaver/default.nix#L74

Reply:
I've followed the example you posted, as I much prefer this over the `let ... in` block.

Done :+1: